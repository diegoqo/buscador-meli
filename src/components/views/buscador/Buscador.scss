.buscador {
  &__container-resultados {
    padding: 20px;
  }
  &__container-desc {
    text-align: left;
  }
  &__container-imagen-buscador {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    &--img {
      width: 100px;
    }
  }
}

.container-imagen-buscador {

}
