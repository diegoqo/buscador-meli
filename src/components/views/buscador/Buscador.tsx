import React, { useEffect } from 'react';
import Helmet from 'react-helmet';
import { Box, Grid } from '@mui/material';
import './Buscador.scss'
import CajaBusqueda from '../../common/cajabusqueda/CajaBusqueda';

const logo = require('../../../images/icon-services-fcs.png');

const Buscador = () => {

    useEffect(() => {
        localStorage.clear();
    }, []);


    return (
        <Box className={'buscador'} sx={{
            width: 1,
        }}>
            <>
                <Helmet>
                    <title>Busquedas</title>
                    <meta name="description" content="Página de búsquedas"/>
                    <meta name="robots" content="INDEX,FOLLOW"/>
                </Helmet>
                <CajaBusqueda key={'caja-busqueda'}/>
                <Grid className={'buscador__container-resultados'} container spacing={2}>
                    <Grid className={'buscador__container-imagen-buscador'} item xs={4}>
                        <img alt={'imagen-buscador'} className={'buscador__container-imagen-buscador--img'} src={logo}/>
                    </Grid>
                    <Grid className={'buscador__container-desc'} item xs={8}><h3>Escribí en el buscador lo que querés
                        encontrar.
                    </h3>
                        <ul>
                            <li><strong>Escribí tu búsqueda</strong> en el campo que figura en la parte superior de
                                la
                                pantalla.
                            </li>
                            <li><a href="/"> Navegá por categorías de productos</a> para encontrar el producto que
                                buscás.
                            </li>
                        </ul>
                    </Grid>
                </Grid>
                <Box>
                </Box>
            </>
        </Box>
    );
}

export default Buscador;


