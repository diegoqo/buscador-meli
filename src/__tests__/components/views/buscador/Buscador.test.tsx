import Buscador from '../../../../components/views/buscador/Buscador';
import { cleanup, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

describe('Probando el componente Buscador', () => {

    it('deberÃ­a mostrar Buscador correctamente', () => {
        const { asFragment } = render(<BrowserRouter><Buscador/></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    })
})
