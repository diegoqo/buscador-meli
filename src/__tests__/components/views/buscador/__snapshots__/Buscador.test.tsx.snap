// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente Buscador debería mostrar Buscador correctamente 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 container-caja-busqueda css-125mmfg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 container-img-buscador css-gj1fbr-MuiGrid-root"
      >
        <img
          src="https://http2.mlstatic.com/frontend-assets/ui-navigation/5.18.8/mercadolibre/logo__small@2x.png"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 container-input-buscador css-vok43g-MuiGrid-root"
      >
        <form
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 paper-input-buscador css-1ps6pg7-MuiPaper-root"
        >
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary input-buscador css-1le28he-MuiInputBase-root"
            data-testid="input-busqueda"
          >
            <input
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Buscar productos, marcas y más..."
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 container-resultados css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 container-imagen-buscador css-gj1fbr-MuiGrid-root"
      >
        <img
          alt="imagen-buscador"
          class="img-buscador"
          src="icon-services-fcs.png"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 container-desc-buscador css-vok43g-MuiGrid-root"
      >
        <h3>
          Escribí en el buscador lo que querés encontrar.
        </h3>
        <ul>
          <li>
            <strong>
              Escribí tu búsqueda
            </strong>
             en el campo que figura en la parte superior de la pantalla.
          </li>
          <li>
            <a
              href="/"
            >
               Navegá por categorías de productos
            </a>
             para encontrar el producto que buscás.
          </li>
        </ul>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</DocumentFragment>
`;
